packages/graphile-utils/src/makeChangeNullabilityPlugin.ts

```diff
@@ -1,49 +1,121 @@
 import type {
-  SchemaBuilder,
   Build,
   Context,
-  Plugin,
   Options,
+  Plugin,
+  SchemaBuilder,
 } from "graphile-build";
-import type { GraphQLInputFieldConfig, GraphQLFieldConfig } from "graphql";
+import {
+  GraphQLFieldConfig,
+  GraphQLInputFieldConfig,
+  GraphQLInputType,
+  GraphQLNonNull,
+  GraphQLType,
+} from "graphql";
+import { GraphQLList } from "graphql/type/definition";
+
+interface NullabilityOpts {
+  type?: boolean | boolean[];
+  args?: {
+    [argName: string]: boolean | boolean[];
+  };
+}
 
 interface ChangeNullabilityRules {
   [typeName: string]: {
-    [fieldName: string]: boolean;
+    [fieldName: string]: boolean | boolean[] | NullabilityOpts;
   };
 }
 
 export default function makeChangeNullabilityPlugin(
   rules: ChangeNullabilityRules
 ): Plugin {
   return (builder: SchemaBuilder, _options: Options) => {
+    function applyNullability<Type extends GraphQLType | GraphQLInputType>(
```

Comment:
[`GraphQLType` covers input types too](https://github.com/graphql/graphql-js/blob/6a5f51f3bd07f3914c7f1ec01908e9e325b77d58/src/type/definition.ts#L52-L68)

```suggestion
    function applyNullability<Type extends GraphQLType>(
```

Reply:
✅